app/code/Magento/Eav/Setup/EavSetup.php

```diff
@@ -622,6 +622,10 @@ public function getAttributeGroup($entityTypeId, $setId, $id, $field = null)
                 $id = $this->defaultGroupIdAssociations[$id];
             } else {
                 $searchField = 'attribute_group_code';
+                if($id == 'images') {   
```

Comment:
Such change is not acceptable. `Eav` module shouldn't be aware of `Catalog`.

Why we cannot simply change
```
$eavSetup->addAttribute(
                \Magento\Catalog\Model\Product::ENTITY,
                'product_image_attribute',
                [
                    'type' => 'varchar',
                    'label' => 'Product Image Attribute',
                    'input' => 'media_image',
                    'frontend' => 'Magento\Catalog\Model\Product\Attribute\Frontend\Image',
                    'global' => \Magento\Eav\Model\Entity\Attribute\ScopedAttributeInterface::SCOPE_STORE,
                    'filterable' => false,
                    'visible_on_front' => false,
                    'used_in_product_listing' => true,
                    'required' => false,
                    'group' => 'Images'
                ]
            );
```
so that it works in current circumstances?

Reply:
> Such change is not acceptable. `Eav` module shouldn't be aware of `Catalog`.
> 
> Why we cannot simply change
> 
> ```
> $eavSetup->addAttribute(
>                 \Magento\Catalog\Model\Product::ENTITY,
>                 'product_image_attribute',
>                 [
>                     'type' => 'varchar',
>                     'label' => 'Product Image Attribute',
>                     'input' => 'media_image',
>                     'frontend' => 'Magento\Catalog\Model\Product\Attribute\Frontend\Image',
>                     'global' => \Magento\Eav\Model\Entity\Attribute\ScopedAttributeInterface::SCOPE_STORE,
>                     'filterable' => false,
>                     'visible_on_front' => false,
>                     'used_in_product_listing' => true,
>                     'required' => false,
>                     'group' => 'Images'
>                 ]
>             );
> ```
> 
> so that it works in current circumstances?

Hi @orlangur 

The `'group' => 'Images'` expects `attribute_group_name` which fails in case of `Images` group because the function `getAttributeGroup($entityTypeId, $setId, $id, $field = null)` is finding the `attribute_group_name` in `attribute_group_code`.
Hence , i added it to search it by `attribute_group_name` in case of `Images` group.

If we add ` 'group' => 'image-management'` , it will create fine because `image-management` is correct `attribute_group_name` but `'group' => 'Images'` expects `attribute_group_name` so i can not simply change it.

Your suggestions are welcome.

Thank you.

Reply:
@irajneeshgupta thanks, then there is nothing to fix here.